reframe/core/schedulers/flux.py

```diff
@@ -0,0 +1,187 @@
+# Copyright 2016-2022 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+#
+# Flux-Framework backend
+#
+# - Initial version submitted by Vanessa Sochat,
+#   Lawrence Livermore National Lab
+#
+
+import itertools
+import os
+import time
+
+import reframe.core.runtime as rt
+from reframe.core.backends import register_scheduler
+from reframe.core.exceptions import JobError
+from reframe.core.schedulers import JobScheduler, Job
+
+# Just import flux once
+try:
+    import flux
+    import flux.job
+    from flux.job import JobspecV1
+except ImportError:
+    error = "no flux Python bindings found"
+else:
+    error = None
+
+waiting_states = ["QUEUED", "HELD", "WAITING", "PENDING"]
+
+
+class _FluxJob(Job):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self._cancelled = False
+
+        # This is set by the scheduler when both the job's state is
+        # 'COMPLETED' and the job's stdout and stderr are written back
+        self._completed = False
+        self.create_job(**kwargs)
+
+    def create_job(self, **kwargs):
```

Comment:
I would probably not use `kwargs` at all here, since you don't need them in this function. They're good and evil at the same time, so in principle I use them only when needed.